pyanaconda/modules/storage/platform.py

```diff
@@ -167,7 +167,9 @@ def stage1_descriptions(self):
     def stage1_constraints(self):
         """The platform-specific constraints for the stage1 device."""
         constraints = {
-            PLATFORM_DEVICE_TYPES: ["disk"]
```

Comment:
OK I think this is wrong, disks and mdarrays are treated seperately here.

mdarrays on disks are disks for anaconda, so "disk" can be implicitely mdarray Ithink think. 

So we need to set maybe PLATFORM_DEVICE_TYPES: ["partition", "mdarray"]?